pkg/forwarder/forwarder.go

```diff
@@ -346,3 +387,98 @@ func (f *DefaultForwarder) SubmitV1Intake(payload Payloads, extra http.Header) e
+	transactionsIntakeProcesses.Add(1)
+
+	return f.submitProcessLikePayload(processesEndpoint, payload, extra, true)
+}
+
+// SubmitRTProcessChecks sends real time process checks
+func (f *DefaultForwarder) SubmitRTProcessChecks(payload Payloads, extra http.Header) (chan Response, error) {
+	transactionsIntakeRTProcesses.Add(1)
+
+	return f.submitProcessLikePayload(rtProcessesEndpoint, payload, extra, false)
+}
+
+// SubmitContainerChecks sends container checks
+func (f *DefaultForwarder) SubmitContainerChecks(payload Payloads, extra http.Header) (chan Response, error) {
+	transactionsIntakeContainer.Add(1)
+
+	return f.submitProcessLikePayload(containerEndpoint, payload, extra, true)
+}
+
+// SubmitRTContainerChecks sends real time container checks
+func (f *DefaultForwarder) SubmitRTContainerChecks(payload Payloads, extra http.Header) (chan Response, error) {
+	transactionsIntakeRTContainer.Add(1)
+
+	return f.submitProcessLikePayload(rtContainerEndpoint, payload, extra, false)
+}
+
+// SubmitConnectionChecks sends connection checks
+func (f *DefaultForwarder) SubmitConnectionChecks(payload Payloads, extra http.Header) (chan Response, error) {
+	transactionsIntakeConnections.Add(1)
+
+	return f.submitProcessLikePayload(connectionsEndpoint, payload, extra, true)
+}
+
+// SubmitPodChecks sends pod checks
+func (f *DefaultForwarder) SubmitPodChecks(payload Payloads, extra http.Header) (chan Response, error) {
+	transactionsIntakePod.Add(1)
+
+	return f.submitProcessLikePayload(podEndpoint, payload, extra, true)
+}
+
+func (f *DefaultForwarder) submitProcessLikePayload(ep endpoint, payload Payloads, extra http.Header, retryable bool) (chan Response, error) {
+	transactions := f.createHTTPTransactions(ep, payload, false, extra)
+
+	results := make(chan Response, len(transactions))
+	internalResults := make(chan Response, len(transactions))
+	expectedResponses := len(transactions)
+
+	for _, txn := range transactions {
+		txn.retryable = retryable
+		txn.attemptHandler = func(transaction *HTTPTransaction) {
+			if v := transaction.Headers.Get("X-DD-Agent-Timestamp"); v == "" {
+				transaction.Headers.Set("X-DD-Agent-Timestamp", strconv.Itoa(int(transaction.GetCreatedAt().Unix())))
+			}
+
+			if v := transaction.Headers.Get("X-DD-Agent-Attempts"); v == "" {
+				transaction.Headers.Set("X-DD-Agent-Attempts", "1")
+			} else {
+				attempts, _ := strconv.ParseInt(v, 10, 0)
+				transaction.Headers.Set("X-DD-Agent-Attempts", strconv.Itoa(int(attempts+1)))
+			}
+		}
+
+		txn.completionHandler = func(transaction *HTTPTransaction, statusCode int, body []byte, err error) {
+			internalResults <- Response{
+				Domain:     transaction.Domain,
+				Body:       body,
+				StatusCode: statusCode,
+				Err:        err,
+			}
+		}
+	}
+
+	go func() {
+		receivedResponses := 0
+		for {
+			select {
+			case r := <-internalResults:
+				results <- r
+				receivedResponses++
+				if receivedResponses == expectedResponses {
+					close(results)
```

Comment:
Should we not return where ? Seems like we're going to run the loop until the timeout otherwise and double close a channel (which will panic).